### ISSUE TITLE
Improve client side caching reset

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/Atlas#447]
### Expected behavior
After changing source values in WebAPI, client should be able to get new values from server.

### Actual behavior
Settings are cached, and there's no clear way to clear the cache.

### Steps to reproduce behavior
This has been reported on a couple forum threads:
http://forums.ohdsi.org/t/atlas-data-source-issue/3232
http://forums.ohdsi.org/t/fresh-installation-using-broadsea/2046/10

We've also had this type of problem internally.  Server-side caching would be ideal because it takes the burden of every user having to remember how to clear the cache, but at the very least there should be a UI to refresh the sources.

**Comment by PRijnbeek** [comment]
For me this new feature is not working. Not sure this is postgresql related?

The only way for me to get the new configuration showing up in Atlas is to Reload the WebAPI

**Comment by fdefalco** [comment]
I could use more information on this issue.  But let me share first:

There are two caching mechanisms at play - the WebAPI caches the sources that it believes are available based on the source and source_daimon tables.  If you edit these tables after the WebAPI is running you would need to restart the WebAPI or refresh the WebAPI cache by calling the following route:

/WebAPI/source/refresh

Once the WebAPI has the update information from the table then there is client side caching as well where ATLAS caches the sources for 24 hours.  To clear the client's local cache you could use the button that was added and discussed in this thread.  

So, to summarize, changes are first made in the table, then they need to be loaded and changed in the WebAPI and finally in the client.

**Comment by PRijnbeek** [comment]
What information would you like?

Good to understand this double caching mechanism, could I have found this somewhere? As a naive user though you would expect that when I press the button 'Clear Configuration Cache' or maybe call it 'Refresh'? it would update both caching mechanisms. Probably a stupid question, but why can that button not also do that call '/WebAPI/source/refresh'  first and then clear the client's local cache?

**Comment by fdefalco** [comment]
The source/refresh route is intended to be an administrative function as it clears the WebAPI cache and then causes a reload from the database.  I would rather it not be triggered by the clear cache button by the user.  This button is only intended to clear the local browser cache of the information it has stored about the WebAPI.  The plan is to add a button to the administration interface for clearing the cache inside the WebAPI itself.

**Comment by eduard93** [comment]
Resolved the same issue by running:

```
DELETE FROM webapi.achilles_cache
```
